hikari/impl/shard.py

```diff
@@ -530,6 +530,12 @@ async def get_user_id(self) -> snowflakes.Snowflake:
             raise RuntimeError("user_id was not known, this is probably a bug")
         return self._user_id
 
+    def user_id(self) -> snowflakes.Snowflake:
+        if not self._user_id:
+            raise RuntimeError("User ID not known yet")
+
+        return self._user_id
```

Comment:
Been thinking about this. Can we have an alive check here and ensure it is always there?

Reply:
Is that notjust achieving the same thing but with extra steps

Reply:
We could then deprecate and subsequently remove the need for having an async and sync version of the function

Reply:
I'm not exactly sure what you're getting at but the sync method doesn't let you wait until the user id has been set like get_user_id, its just being added so runtime code (the event manager) can get the user id for a shard without having to enter an async context

Reply:
Yes, I understood the idea behind the addition. What I was saying was that they could be merged into one if we can ensure that the user id will always be available if the shard is alive

Reply:
isn't that already guaranteed by both these methods though, just get_user_id yields to the event loop if you call if before the shard's been started  while user_id errors